dex-it/src/test/scala/com/wavesplatform/it/sync/smartcontracts/AllowedBlockchainStateAccountsTestSuite.scala

```diff
@@ -0,0 +1,123 @@
+package com.wavesplatform.it.sync.smartcontracts
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.dex.api.http.entities.HttpOrderStatus
+import com.wavesplatform.dex.domain.account.KeyPair
+import com.wavesplatform.dex.domain.asset.Asset.Waves
+import com.wavesplatform.dex.domain.bytes.ByteStr
+import com.wavesplatform.dex.domain.order.OrderType
+import com.wavesplatform.dex.error.{AccountScriptDeniedOrder, AccountScriptReturnedError}
+import com.wavesplatform.dex.it.test.Scripts
+import com.wavesplatform.it.MatcherSuiteBase
+
+import scala.concurrent.duration._
+
+final class AllowedBlockchainStateAccountsTestSuite extends MatcherSuiteBase {
+
+  "AllowedBlockchainStateAccountsTestSuite" - {
+
+    "should fail while accessing blockchain state from usual accounts" in {
+      updateAccountScript(bob, script1)
+      val order = mkOrder(bob, wavesUsdPair, OrderType.SELL, 10.waves, 1, version = 2)
+      dex1.tryApi.place(order) should failWith(
+        AccountScriptReturnedError.code,
+        "The account's script of 3Q6ujVDbX57oLsXxifqfTcycgb4S8U3DLFz returned the error: An access to the blockchain.height is denied on DEX"
+      )
+    }
+
+    "should succeed while accessing blockchain state from custom accounts" in {
```

Comment:
"from custom accounts" to "from allowed accounts"?

Reply:
fxd